src/basic/socket-util.c

```diff
@@ -235,23 +235,39 @@ int socket_address_parse_and_warn(SocketAddress *a, const char *s) {
 }
 
 int socket_address_parse_netlink(SocketAddress *a, const char *s) {
-        int family;
+        _cleanup_free_ char *word = NULL;
         unsigned group = 0;
-        _cleanup_free_ char *sfamily = NULL;
+        int family, r;
+
         assert(a);
         assert(s);
 
         zero(*a);
         a->type = SOCK_RAW;
 
-        errno = 0;
-        if (sscanf(s, "%ms %u", &sfamily, &group) < 1)
-                return errno > 0 ? -errno : -EINVAL;
+        r = extract_first_word(&s, &word, NULL, 0);
+        if (r < 0)
+                return r;
+        if (r == 0)
+                return -EINVAL;
 
-        family = netlink_family_from_string(sfamily);
+        family = netlink_family_from_string(word);
         if (family < 0)
                 return -EINVAL;
 
+        word = mfree(word);
+        r = extract_first_word(&s, &word, NULL, 0);
```

Comment:
hmm, for the second word, we can just use `s` directly; after all, there can only be one word left.  Hence, instead of calling into extract_first_word() a second time, let's just use `s` right away. (note that extract_first_word() will not just return the first word, but advance `s` to the next word, skipping whitespace between the two words, hence this is perfect for this usecase)

Reply:
It sounds simpler, but currently it supports trailing white spaces. Simply passing the remaining string to `safe_atou()` drops the functionality. But I am not sure whether such a functionality is necessary or not. I will check all usecases later. Thanks.